packages/node-core/src/indexer/storeCache/cacheModel.ts

```diff
@@ -67,22 +77,25 @@ export class CachedModel<
     options: {offset?: number; limit?: number} | undefined
   ): Promise<T[] | undefined> {
     let cachedData = this.getByFieldFromCache(field, value);
-    if (options?.offset) {
-      if (cachedData.length <= options.offset) {
-        // example cache length 16, offset is 30
-        // it should skip cache value
-        cachedData = [];
-      } else if (cachedData.length > options.offset + options.limit) {
-        // example cache length 166, offset is 30, limit is 50
-        // then return all from cache [30,80]
-        return cachedData.slice(options.offset, options.offset + options.limit);
-      } else if (cachedData.length < options.offset + options.limit) {
-        // example cache length 66, offset is 30, limit is 50
-        // then return [30,66] from cache, set new limit and join record from db
-        cachedData = cachedData.slice(options.offset, cachedData.length);
-        options.limit = options.limit - (cachedData.length - options.offset);
-      }
+    if (options?.offset === undefined) {
+      options.offset = 0;
```

Comment:
`options` could be undefined.

Seems to be other cases in this function that should also check for options being defined

Reply:
option will be defined here, I think it is best to move this code together

https://github.com/subquery/subql/blob/b9a9c860f45ccb7a25ec793a66db31932bcbb020/packages/node-core/src/indexer/store.service.ts#L721